prover/src/entry.rs

```diff
@@ -34,6 +34,18 @@ impl<const N_CURRENCIES: usize> Entry<N_CURRENCIES> {
         &self.balances
     }
 
+    pub fn concatenated_balance(&self) -> BigUint {
+        let mut concatenated_balance = BigUint::from(0u32);
+
+        // To arragne the balances in the correct order, we need to reverse the array
+        for (i, balance) in self.balances.iter().rev().enumerate() {
+            // Shift bits: 1 for buffer; 19 is highest bit for two power of userbase; 64 is for the maximum range check limit
+            concatenated_balance += balance << ((1 + 19 + 64) * i);
+        }
```

Comment:
I think it's better to have `19` as a constant or a generic and also add an assertion that the concatenated balance would not exceed the field bit size.